harness/determined/common/experimental/project.py

```diff
@@ -0,0 +1,214 @@
+
+    def archive(self) -> None:
+        """Set the project to archived (archived = True).
+
+        As with other setters, this will change the attribute both on master and this local object.
+        """
+        bindings.post_ArchiveProject(session=self._session, id=self.id)
+        self.archived = True
+
+    def unarchive(self) -> None:
+        """Set the project to unarchived (archived = False).
+
+        As with other setters, this will change the attribute both on master and this local object.
+        """
+        bindings.post_UnarchiveProject(session=self._session, id=self.id)
+        self.archived = False
+
+    def add_note(self, name: str, contents: str) -> None:
+        """Add a note to the project.
+
+        Because there is not yet functionality on the backend to add a single note, this method:
+        1. fetches current notes for this project from the master.
+        2. adds the new note to the list of notes.
+        3. sends the updated list of notes to the master.
+
+        WARNING:
+        When it exits, the notes attribute of this object will be the same notes updated on master.
+        This set of notes might reflect changes to the project that have happened since the project
+        was last hydrated from master.
+
+        Args:
+            name: The name of the note.
+            contents: The contents of the note.
+        """
+        master_notes = bindings.get_GetProject(session=self._session, id=self.id).project.notes
+        combined_notes = list(master_notes) + [bindings.v1Note(name=name, contents=contents)]
+
+        request_body = bindings.v1PutProjectNotesRequest(notes=combined_notes, projectId=self.id)
+        resp = bindings.put_PutProjectNotes(
+            session=self._session, body=request_body, projectId=self.id
+        )
+
+        self.notes = [note.to_json() for note in resp.notes]
+
+    def remove_note(self, name: str) -> None:
+        """Remove a note from the project.
+
+        Because there is not yet functionality on the backend to remove a single note, this method:
+        1. fetches current notes for this project from the master.
+        2. removes the note with the passed name from the list of notes.
+        3. sends the updated list of notes to the master.
+
+        WARNING:
+        When it exits, the notes attribute of this object will be the same notes updated on master.
+        This set of notes might reflect changes to the project that have happened since the project
+        was last hydrated from master.
+
+        Args:
+            name: The name of the note to remove.
+
+        Raises:
+            ValueError: If no note with the passed name is found
+            NotImplementedError: If multiple notes with the passed name are found
+
+        It is possible to for a project to have multiple notes with the same name. This method's
+        interface doesn't make it possible to choose which of them to remove.
+        """
```

Comment:
"""Remove a note from the project by name.
Because there is no functionality on the backend to remove a single note, this method fetches current notes, removes the note, and updates the master.
WARNING:
Post-execution, the object's notes attribute matches the updated master's notes,
possibly reflecting changes to the project that have happened since the project was last hydrated from master.

Args:
name: Title of the note to remove.

        Raises:
            ValueError: If the note with the passed name isn't found
            NotImplementedError: If multiple notes with the passed name are found
        It is possible to for a project to have multiple notes with the same name. The method's
        interface doesn't make it possible to choose which note to remove.
        """

Reply:
I've changed this method's interface. Could you please take a look?

Reply:
i'll come back to this one

Reply:
looks good. setting to resolved.